File: src/main/java/infinitespire/relics/EvilPickle.java
package infinitespire.relics;

import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import infinitespire.InfiniteSpire;
import infinitespire.abstracts.Relic;
import infinitespire.rewards.InterestReward;

public class EvilPickle extends Relic {
	public static final String ID = InfiniteSpire.createID("EvilPickle");

	public EvilPickle(){
		super(ID, "evilpickle", RelicTier.SPECIAL, LandingSound.FLAT);
	}

	@Override
	public void onTrigger() {
		this.flash();
		AbstractDungeon.combatRewardScreen.rewards.add(new InterestReward(Math.round(AbstractDungeon.player.gold * 0.065f)));
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/relics/EvilPickle.kt
package com.blanktheevil.infinitespire.relics

import basemod.AutoAdd
import com.blanktheevil.infinitespire.extensions.makeID
import com.blanktheevil.infinitespire.extensions.player
import com.blanktheevil.infinitespire.rewards.InterestReward
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import kotlin.math.round

@AutoAdd.Ignore
class EvilPickle : Relic(ID, IMG, TIER, SOUND) {
  companion object {
    val ID = "EvilPickle".makeID()
    private const val IMG = "evilpickle"
    private val TIER = RelicTier.SPECIAL
    private val SOUND = LandingSound.FLAT
  }

  override fun onTrigger() {
    this.flash()
    AbstractDungeon.combatRewardScreen.rewards.add(
      InterestReward(
        round(player.gold.times(0.065f)).toInt()
      )
    )
  }
}